src/lucky/routeable.cr

```diff
@@ -19,16 +19,39 @@ module Lucky::Routeable
     # **See also** our guides for more information and examples:
     # * [Routing](https://luckyframework.org/guides/actions-and-routing/#routing)
     macro {{ http_method.id }}(path)
-      \{% unless path.starts_with?("/") %}
-        \{% path.raise "Path must start with a slash. Example: '/" + path + "'" %}
-      \{% end %}
-
-      add_route :{{ http_method.id }}, \{{ path }}, \{{ @type.name.id }}
-
-      setup_call_method(\{{ yield }})
+      match(:{{ http_method.id }}, \{{ path }}) do
+        \{{ yield }}
```

Comment:
This is a bit strange but I can’t seem to figure out how to forward blocks from macros. I’m quite new to Crystal.

Reply:
Macros are tricky. This is how I've done it in the past and I'm not aware of something better. So I think this is good!